src/Web/Controllers/AccountController.cs

```diff
@@ -103,120 +101,166 @@ public async Task<ActionResult> SignOut()
+                MarketingContext = marketingContext,
+                StoreFrontContext = storefrontContext,
+                DeviceContext = deviceContext,
+            };
+
+            var correlationId = _fraudProtectionService.NewCorrelationId;
+
+            var signupAssessment = await _fraudProtectionService.PostSignup(signupEvent, correlationId);
+
+            //Track Fraud Protection request/response for display only
+            var fraudProtectionIO = new FraudProtectionIOModel(signupEvent, signupAssessment, "Signup");
+
+            //2 out of 3 signups will succeed on average. Adjust if you want more or less signups blocked for tesing purposes.
+            var random = new Random();
+            var rejectSignup = random.NextDouble() >= 2.0 / 3;
+            var signupStatusType = rejectSignup ? SignupStatusType.Rejected.ToString() : SignupStatusType.Approved.ToString();
+
+            var signupStatus = new SignupStatusEvent
+            {
+                SignUpId = signupEvent.SignUpId,
+                StatusType = signupStatusType,
+                StatusDate = DateTimeOffset.Now,
+                Reason = "User is " + signupStatusType
+            };
+
+            if (!rejectSignup)
+            {
+                signupStatus.User = new SignupStatusUser { UserId = model.Email };
+            }
+
+            var signupStatusResponse = await _fraudProtectionService.PostSignupStatus(signupStatus, correlationId);
+
+            fraudProtectionIO.Add(signupStatus, signupStatusResponse, "Signup Status");
+
+            TempData.Put(FraudProtectionIOModel.TempDataKey, fraudProtectionIO);
+            #endregion
+
+            if (rejectSignup)
+            {
+                ModelState.AddModelError("", "Signup rejected by Fraud Protection. You can try again as it has a random likelyhood of happening in this sample site.");
+                return View(model);
+            }
+
+            //Only create the user in the sample site if the Fraud Protection merchant decision is APPROVE
+            var user = new ApplicationUser
+            {
+                UserName = model.Email,
+                Email = model.Email,
+                FirstName = model.FirstName,
+                LastName = model.LastName,
+                PhoneNumber = model.Phone,
+                Address1 = model.Address1,
+                Address2 = model.Address2,
+                City = model.City,
+                State = model.State,
+                ZipCode = model.ZipCode,
+                CountryRegion = model.CountryRegion
+            };
+
+            var result = await _userManager.CreateAsync(user, model.Password);
+            if (!result.Succeeded)
+            {
                 AddErrors(result);
+                return View(model);
             }
 
-            // If we got this far, something failed, redisplay form
-            return View(model);
+            if (user == null)
+                throw new Exception(nameof(user));
```

Comment:
need { } :)

Reply:
also how would user ever be null?

Reply:
Not sure!  Thanks for pointing out.  I've removed it.